web/src/routes/SearchPage.tsx

```diff
@@ -38,6 +38,11 @@ const SearchPage = ({ city, cityCode, languageCode, pathname }: CityRouteProps):
   const navigate = useNavigate()
   const fallbackLanguage = config.sourceLanguage
 
+  useEffect(() => {
+    const appendToUrl = filterText.length !== 0 ? `?query=${filterText}` : ''
+    navigate(`${pathname}/${appendToUrl}`, { replace: true })
+  }, [filterText, navigate, pathname])
+
```

Comment:
:wrench: Do you think this `useEffect` is really needed? What do you think of setting the query to the `languageChangePaths` in L68?

Reply:
you are right, it makes more sense just to include it here :+1: